packages/flutter_test/lib/src/matchers.dart

```diff
@@ -277,6 +278,144 @@ Matcher matchesGoldenFile(dynamic key) {
   throw new ArgumentError('Unexpected type for golden file: ${key.runtimeType}');
 }
 
+/// Asserts that a [SemanticsData] contains the specified information.
+///
+/// If the either the label, hint, value, textDirection, or rect fields are not
+/// provided, then they are not part of the comparison.  All of the boolean
+/// flag and action fields must match, and default to false.
+///
+/// To retrieve the semantics data of a widget, use [tester.getSemanticsData]
+/// with a [Finder] that returns a single widget.  Semantics must be enabled
```

Comment:
What error do you get if you forget to enable semantics? Is it clear how you can fix that error?

Reply:
Updated the getSemanticsData to throw a state error which tells you to enable semantics